core/src/health.rs

```diff
@@ -8,7 +8,9 @@ use std::sync::atomic::{AtomicBool, Ordering};
 /// Trait for asyncronously notifying health information.
 #[cfg_attr(test, mockall::automock)]
 pub trait HealthReporting: Send + Sync {
-    /// Notify that the service is ready.
+    /// Notify that the service is ready. Can be called multiple times.
+    /// We use this to signal healthiness only at the start of a batch in order to not interrupt the
```

Comment:
nit: we signal readiness not healthiness. Not sure we need to document what this is used for (as this is is a detail that might change but the comment likely won't).

Reply:
To me it's useful when I imagine someone later looking at this and wondering what the point of the health reporting trait is.